MapboxDirections/MBVisualInstructionDetailComponent.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+
+#if os(iOS) || os(tvOS)
+    import UIKit
+#endif
+
+
+/**
+ :nodoc:
+ Contains necessary features to build a more rich visual experience for a `RouteStep`.
+ */
+@objc(MBVisualInstructionComponent)
+public class VisualInstructionComponent: NSObject, NSSecureCoding {
+    
+    /**
+     :nodoc:
+     Single part of a visual instruction.
+     */
+    public let text: String?
+    
+    /**
+     :nodoc:
+     Dictionary containing `UITraitCollection` for scales 1.0, 2.0 and 3.0. Each key's value is a `URL`.
+     */
+    #if os(iOS) || os(tvOS)
+        public var imageURLS: [UITraitCollection: URL] = [:]
+    #else
+        public var imageURLS: [NSNumber: URL] = [:]
+    #endif
+    
+    internal init(json: JSONDictionary) {
+        text = json["text"] as? String
+        
+        if let baseURL = json["imageBaseURL"] as? String {
+            let oneXURL = URL(string: "\(baseURL)@1x.png")
+            let twoXURL = URL(string: "\(baseURL)@2x.png")
+            let threeXURL = URL(string: "\(baseURL)@3x.png")
+            
+            #if os(iOS) || os(tvOS)
+                imageURLS[UITraitCollection(displayScale: 1.0)] = oneXURL
+                imageURLS[UITraitCollection(displayScale: 2.0)] = twoXURL
+                imageURLS[UITraitCollection(displayScale: 3.0)] = threeXURL
```

Comment:
Upon seeing this code, I have to wonder: do we need to store the URLs for all three scale factors if the current screen can only have one scale factor at a time? It would be simpler to store just one image URL computed using the same logic seen here:

https://github.com/mapbox/MapboxStatic.swift/blob/279ce127c788f0bcaee15e214a1bee433e22eeca/MapboxStatic/SnapshotOptions.swift#L220-L247

macOS applications would be unable to account for changes to the scale factor, such as when the window containing the component gets dragged from a LoDPI screen to a HiDPI screen mid-route. But that’s multiple levels of edge case that we hopefully aren’t worrying about right now.

The only other reason I can think of for keeping track of multiple scale factors is that the navigation SDK might choose to display a lower-resolution image while on a poor connection and a best-resolution image while on a good connection. But these images are so small that I can’t imagine the scale factor mattering much even then.